src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+SurrealDB will automatically convert the idiom part `.uppercase()` to the function `string::uppercase()` when the query is executed.
+
+To learn more about string method chaining in SurrealQL, see the [string functions](/docs/surrealql/functions/database/string#method-chaining) section.
+
+### Graph Navigation
+
+SurrealDB can also be used in the context of graph databases, where data is stored and navigated using graph traversal idioms. The [`RELATE` statement](/docs/surrealql/statements/relate) is used to create relationships between records. This allows you to traverse related records efficiently without needing to pull data from multiple tables and merging that data together using SQL JOINs.
+
+
+For example, let's consider the following data:
+
+```surql title="Create a new planet, city, and explorer records"
+CREATE planet:unknown_planet;
+CREATE city:el_dorado          SET name = "El Dorado";
+CREATE explorer:drake          SET name = "Drake";
+CREATE explorer:local_guide    SET name = "Local Guide";
+
+RELATE explorer:drake->discovered->planet:unknown_planet;
+RELATE explorer:drake->visited->city:el_dorado;
+RELATE explorer:local_guide->assisted->explorer:drake;
+
+```
+
+```surql title="Retrieve all relationships from Drake"
+SELECT 
+    *,
+    ->? AS actions, 
+    <-? AS interactions, 
+    <->? AS connections
+FROM explorer:drake;
+```
+
+```json title="Response"
+[
+	{
+		actions: [
+			discovered:sh9zbsz5u705cxv6qgoi,
+			visited:hmtttiqqfa4mt9is1a7j
+		],
+		connections: [
+			assisted:1pv8k3p1wpuf0guf5bvm,
+			discovered:sh9zbsz5u705cxv6qgoi,
+			visited:hmtttiqqfa4mt9is1a7j
+		],
+		id: explorer:drake,
+		interactions: [
+			assisted:1pv8k3p1wpuf0guf5bvm
+		],
+		name: 'Drake'
+	}
+]
+```
+
+Explanation:
+
+- `*`: Selects all fields of `explorer:drake`.
+- `->? AS actions`: Retrieves all outgoing relationships from Drake and aliases them as actions.
+- `<-? AS interactions`: Retrieves all incoming relationships to Drake and aliases them as interactions.
+- `<->? AS connections`: Retrieves all relationships connected to Drake, regardless of direction, and aliases them as connections.
+
+
+### Destructuring
+
+When working with nested data, you can destructure objects using the `.` and `{ ... }` idioms.
+
+For example, 
+
+```surql title="Create a new person record"
+CREATE person:1 SET name = 'John', age = 21, obj = { a: 1, b: 2, c: { d: 3, e: 4, f: 5 } };
+```
+
+```json title="Response"
+[
+	{
+		age: 21,
+		id: person:1,
+		name: 'John',
+		obj: {
+			a: 1,
+			b: 2,
+			c: {
+				d: 3,
+				e: 4,
+				f: 5
+			}
+		}
+	}
+]
+```
+
+```surql
+SELECT obj.{ a, c.{ e, f } } FROM person;
+```
+
+```json title="Response"
+[
+	{
+		obj: {
+			a: 1,
+			c: {
+				e: 4,
+				f: 5
+			}
+		}
+	}
+]
+```
+You can also OMIT fields that you don't want to destructure using the `OMIT` clause.
+
+```surql 
+SELECT * OMIT obj.c.{ d, f } FROM person;
+```
+
+
+```json title="Response"
+[
+	{
+		a: 1,
+		c: {
+			e: 4
+		}
+	}
+]
+```
+
```

Comment:
My favourite part of destructuring/restructuring is how it works with graph queries. If you want you could add the following query to riff off of the Drake example:

SELECT ->visited->city.{name, id}
FROM explorer:drake;